homeassistant/components/radio_browser/media_source.py

```diff
@@ -49,7 +49,9 @@ def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
     @property
     def radios(self) -> RadioBrowser | None:
         """Return the radio browser."""
-        return self.hass.data.get(DOMAIN)
+        if self.entry.runtime_data:
+            return self.entry.runtime_data
+        return None
```

Comment:
Just wondering: Won't `self.entry.runtime_data` always be a `RadioBrowser` instance?

Reply:
i'm not sure if it is possible, that `media_source.async_get_media_source` is called before `__init__.async_setup_entry` is completed?

Reply:
ok - it seems the `component.media_source` platform waits until a integration/component is loaded, before calling `async_get_media_source` in each integration 🤔

Reply:
Might be completely missing something, but I believe we need a loaded integration for platforms to be setup in the first place. There is also the `entry = hass.config_entries.async_entries(DOMAIN)[0]` call in `async_get_media_source` which would error if we don't have a config entry and each entry is created with the `runtime_data` in place 🤷🏻‍♂️